Sources/MapboxMaps/Gestures/GestureRecognizers/MapboxLongPressGestureRecognizer.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import UIKit
+
+internal final class MapboxLongPressGestureRecognizer: UILongPressGestureRecognizer {
+    private var translationReference: CGPoint = .zero
+    private var translation: CGPoint = .zero
+
+    override init(target: Any?, action: Selector?) {
+        super.init(target: target, action: action)
+
+        addTarget(self, action: #selector(stateDidChange))
+    }
+
+    @objc private func stateDidChange() {
+        switch state {
+        case .began:
+            translationReference = location(in: view)
+        case .changed:
+            let touchLocation = location(in: view)
+            translation = CGPoint(
+                x: translationReference.x - touchLocation.x,
+                y: translationReference.y - touchLocation.y
+            )
+        default:
+            break
+        }
+    }
+
+    internal func translation(in view: UIView?) -> CGPoint {
+        if self.view == view {
+            return translation
+        }
+
+        return self.view?.convert(translation, to: view ?? self.view?.window) ?? .zero
+    }
+
+    internal func setTranslation(_ translation: CGPoint, in view: UIView?) {
+        guard let newTranslation = self.view?.convert(translation, from: view) else {
+            return
+        }
+
+        let touchLocation = location(in: self.view)
+
+        self.translationReference = CGPoint(
+            x: touchLocation.x + newTranslation.x,
+            y: touchLocation.y + newTranslation.y
```

Comment:
Should we enforce the check that `translationReference` has the same `view` all the time? 🤔